feat: post files map / gallery view UI

blushi opened the pull request:
## Description

Closes: #2255

<!-- Add a description of the changes that this PR introduces and the files that
are the most critical to review. -->

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [ ] provided a link to the relevant issue or specification
- [ ] provided instructions on how to test
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### How to test

<!-- This should include steps on how to test the fixes or features within the marketplace app and/or the website.
If applicable, this should also include links to the created/updated components on storybook. -->

1.

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please **add
your handle next to the items reviewed if you only reviewed selected items**._

I have...

- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed code correctness and readability
- [ ] verified React components follow DRY principles
- [ ] reviewed documentation is accurate
- [ ] reviewed tests
- [ ] manually tested (if applicable)

blushi commented on tailwind.css:
```diff
@@ -71,5 +72,6 @@
     --blue-300: 177 202 236;
     --blue-200: 216 228 245;
     --blue-100: 61 122 207;
+    --blue-50: 239 244 251; /** replace with correct dark color **/
```
still pending ref: https://www.figma.com/file/Bksz1JeDYxQVIXdI46EgPT?node-id=835:50395&mode=dev#691690703 cc/ @erikalogie

erikalogie replied:
replied in Figma

erikalogie commented:
Great work! This is looking amazing!

Is it possible to make the white area the same height on all the cards on mobile? https://monosnap.com/file/aznlOn6TrJKdtn3xUvH3UvyUEGBNwF

I think that whatever the point is on the map that is highlighted, it should always be visible in the window on mobile, similar to this example:
https://github.com/regen-network/regen-web/assets/8460237/5578c8df-2fc5-4e05-9d7c-86931dea68b2

This shouldn't be possible: https://monosnap.com/file/78rriRBz4mjpPEBwGGwNeX0TGsq2mu

blushi commented:
> Great work! This is looking amazing!
> 
> Is it possible to make the white area the same height on all the cards on mobile? https://monosnap.com/file/aznlOn6TrJKdtn3xUvH3UvyUEGBNwF
> 
> I think that whatever the point is on the map that is highlighted, it should always be visible in the window on mobile, similar to this example: https://github.com/regen-network/regen-web/assets/8460237/5578c8df-2fc5-4e05-9d7c-86931dea68b2
> 
> This shouldn't be possible: https://monosnap.com/file/78rriRBz4mjpPEBwGGwNeX0TGsq2mu

@erikalogie I've addressing your comments.

erikalogie commented:
Great! 

Some improvements:
1) Have the read more link left align with paragraph: https://monosnap.com/file/NBb1QittH5bGc3Oa8KBmOkIwmbbAYE
2) The height of the caption area should be one line high only in the case of the csv file and the video with short description in this video so it is more consistent. Either that, or if there is a 2 line description then all of the caption heights are 2 lines, regardless:
https://github.com/regen-network/regen-web/assets/8460237/e02fe63c-7b33-4819-8db9-6739d86bb6cb

It would be great to have @clevinson look at this too. I'm thinking about a couple UX things that I think could be improved and I'd like some feedback before implementing:
1) I'm wondering if it would be more intuitive if the user clicks anywhere on the image area that it opens the image, document, csv, etc, in a new tab/window. Then we could get rid of the little icon that opens in a new window
2) I'm thinking to have the three different document types (audio file, document, spreadsheet) that we don't have preview for each have a specific color that shows up in both the preview window and the thumbnails along the side. That way it is easier for the user's eye to correlate the two different areas. https://www.figma.com/design/Bksz1JeDYxQVIXdI46EgPT/Project-Posts?node-id=3674%3A87861&t=pHAwvKyo3IMuG1PT-1

blushi commented:
> Some improvements:
> 
> 1. Have the read more link left align with paragraph: https://monosnap.com/file/NBb1QittH5bGc3Oa8KBmOkIwmbbAYE
> 2. The height of the caption area should be one line high only in the case of the csv file and the video with short description in this video so it is more consistent. Either that, or if there is a 2 line description then all of the caption heights are 2 lines, regardless:
>    https://github.com/regen-network/regen-web/assets/8460237/e02fe63c-7b33-4819-8db9-6739d86bb6cb

@erikalogie I've addressed 1 and 2.
If that looks good, I'll merge this and we get to your suggestions in a follow-up PR, does that work?

erikalogie commented:
@blushi great, yes let's maybe just bring this up during the sprint planning today
